src/pages/settings/tabs/UserManagementSettings.tsx

```diff
@@ -133,15 +135,15 @@ function UserManagementSettings() {
 
   const changeAvatar = (avatar: File | undefined) => {
     if (!avatar) return;
-    editAvatar({ avatar, userId: selectedUser.ID })
-      .then(() => toast.success('', 'Avatar updated!'))
+    fileToDataUrl(avatar)
+      .then(dataUrl => setSelectedUser(user => ({ ...user, Avatar: dataUrl })))
       .catch(error => console.error(error));
   };
 
   const removeAvatar = () => {
-    deleteAvatar(selectedUser.ID)
-      .then(() => toast.success('', 'Avatar removed!'))
-      .catch(error => console.error(error));
+    // Setting the avatar to an empty string will tell the server to remove the
+    // avatar.
+    setSelectedUser(user => ({ ...user, Avatar: '' }));
```

Comment:
and now here you do not use deep clone which would keep the community sites object at the same address which is fine I guess, but actually here deepClone would make more sense, then again please use immer for that and don't do it manually